### PR TITLE
enable debug logs for vpc

### LINKED ISSUE #625: Enable debug logs for vpc
/kind feature
/area provider/ibmcloud

**Describe the solution you'd like**
Explore an option to enable debug logs for the vpc clients

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
Signed-off-by: Prajyot-Parab <prajyot.parab2@ibm.com>

**What this PR does / why we need it**:
- Enable debug logs for vpc
- Expose additional spec params

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #625 

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
enable debug logs for vpc and expose additional spec params
```

### CODE REVIEW

**Comment by Prajyot-Parab** [comment]
@mkumatag does this logic seems okay to enable deeper level of debug logs?

File: cloud/scope/cluster.go [hunk 1]
```diff
@@ -76,6 +81,10 @@ func NewClusterScope(params ClusterScopeParams, authenticator core.Authenticator
 		return nil, errors.Wrap(vpcErr, "failed to create IBM VPC session")
 	}
 
+	if core.GetLogger().GetLogLevel() != VPCDebugLevel {
+		core.SetLoggingLevel(VPCDebugLevel)
+	}
+
```
Comment (mkumatag) [inline cloud/scope/cluster.go:87]: it should be like

```golang
if params.Logger.V(DEBUGLEVEL).Enabled():
    core.SetLoggingLevel(core. LevelDebug)
```

>Note: untested code.

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: cloud/scope/cluster.go (hunk 1 above)
Comment (Prajyot-Parab) [reply cloud/scope/cluster.go]: Tested working fine.

File: templates/cluster-template-powervs.yaml [hunk 2]
```diff
@@ -165,6 +165,10 @@ spec:
         name: "${IBMPOWERVS_IMAGE_NAME}"
       network:
         name: "${IBMPOWERVS_NETWORK_NAME}"
+      memory: ${IBMPOWERVS_MEMORY:="4"}
+      processors: ${IBMPOWERVS_PROCESSORS:="0.25"}
+      sysType: ${IBMPOWERVS_SYSTYPE:="s922"}
+      procType: ${IBMPOWERVS_PROCTYPE:="shared"}
```
Comment (mkumatag) [inline templates/cluster-template-powervs.yaml:171]: can these defaults be part of another PR?

File: templates/cluster-template-powervs.yaml [hunk 3]
```diff
@@ -202,6 +206,10 @@ spec:
...
```
Comment (mkumatag) [inline templates/cluster-template-powervs.yaml:212]: can these defaults be part of another PR?

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: templates/cluster-template-powervs.yaml (hunk 3 above)
Comment (Prajyot-Parab) [reply templates/cluster-template-powervs.yaml]: Done

**Review by mkumatag — APPROVED** [review]
/lgtm
/approve
